js/index.js

```diff
@@ -2,41 +2,60 @@
 
 function updateSubtotal(product) {
   console.log('Calculating subtotal, yey!');
-
-  //... your code goes here
+ 
+  const price = product.querySelector('.price span').innerHTML;
+  const quantity = product.querySelector('.quantity input').value;
+  let priceSubtotal = price * quantity;
+  product.querySelector('.subtotal span').innerHTML = priceSubtotal;
+  return priceSubtotal;
+  
+  
 }
 
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
-
-  // ITERATION 2
-  //... your code goes here
+  const allProducts = document.getElementsByClassName('product');
+  let total = 0
+  for (const product of allProducts) {
+    total += updateSubtotal(product);
+  }
+  
+  
 
   // ITERATION 3
-  //... your code goes here
+ document.querySelector('#total-value span').innerHTML = total;
```

Comment:
```suggestion
 document.querySelector('#total-value span').innerText = total;
```
Lo mismo de siempre :D